source/developer/fx-guidelines.rst

```diff
@@ -74,95 +74,106 @@ Design Guidelines
 
-|
----------------------------
+        Error message does not extend beyond the horizontal line separator.
+            ..  image:: ../images/align2.png
+                :alt: Confined messages with respect to width
+                :width: 500 px
 
-Notifications: Header
----------------------
+**Horizontally align multi-line elements along a vertical line.**
 
-Appears as an additional header at the top of the UI. May include an “x” button to dismiss. Limited to 60 characters. May include a time-out. Header notification should appear over the rest of the UI instead of pushing the UI down. Multiple headers can appear, with the earliest notification at the top and more recent notifications appearing underneath.
+    **Example:**
 
-**Desktop**
+        Roles right justified with respect to the text and irrespective of the icon.
+            ..  image:: ../images/align3.jpg
+                :alt: Vertically justified
+                :width: 500 px
 
-    ..  figure:: ../images/header1.png
-        :alt: Header Notification Desktop
+**Instructions should be sentences, one-line links should be fragments.**
 
-**Mobile**
+Instructions, such as “A password reset link has been sent to ``you@email.com`` for your account. Please check your inbox.”, should be displayed as sentences ending in periods. One-line links, such as “Find it here”, should not end in periods or commas, but question marks are okay.
 
-    ..  image:: ../images/header2.png
-        :alt: Header Notification Mobile
+    **Example:**
 
-There are several modes for this element:
+        Incorrect:
+            Instruction “Please check your inbox”, didn't end with a period.
 
-    **System-wide header notification:**
+            ..  image:: ../images/align4.png
+                :alt: Period Missing
+                :width: 300 px
 
-        | Triggered on login. Includes “x” button to dismiss. No timeout. Removed when system admin cancels.
-        |
-        | Examples:
-        | “Try our new Windows App. Click here to download.”,
-        | “Scheduled maintenance 2:00am to 2:30am starts in 20 minutes, 3 seconds.”
+        Correct:
+            Instruction “Please check your inbox”, ended with a period.
 
-    **Action required header notification:**
+            ..  image:: ../images/align5.jpg
+                :alt: Period added
+                :width: 300 px
 
-        | Triggered on login. No “x” button to dismiss. No timeout. Dismissed when action completed.
-        |
-        | Examples:
-        | “We’re evolving. Please review and accept our new terms of service.”
+---------------------------
 
-    **Persistent issue header notification:**
+Display: Button Placement
+-------------------------
 
-        | Triggered on issue. No “x” button to dismiss. Dismissed when issue no longer persists.
-        |
-        | Examples:
-        | “You are not connected to the internet.”
+**Dialog BOTTOM RIGHT BUTTONS should be either “Close”, or “Cancel” and [ACTION_BUTTON].**
```

Comment:
Probably to prioritise it, but I've changed it to:
"Buttons in the footer of the dialog should either be..."